openwisp_monitoring/monitoring/base/models.py

```diff
@@ -115,6 +115,10 @@ def full_clean(self, *args, **kwargs):
         self.key = self._makekey(self.key)
         return super().full_clean(*args, **kwargs)
 
+    @classmethod
+    def post_delete_receiver(cls, instance, *args, **kwargs):
+        timeseries_db.delete_metric_data(key=instance.key, tags=instance.tags)
```

Comment:
The problem with this is that this may fail, moreover if multiple devices are deleted, the likelyhood of failure is greater.

We need to execute this in a background task with some retry mechanism.
It seems to me that this particular method `delete_metric_data`, does not have a built-in retry mechanism, therefore we could simply let the celery task raise the exception on failures and let the celery task retry whenever the exception is raised, as we do for writing the data.

To sum up, I believe deleting data should work in a similar manner to how the data is written, I think we can simply replicate that kind of structure (executed in a dedicated celery task, which has the retry on failure argument specified).

What do you think?

Reply:
I have added a celery task for deletion of the data. The celery task does not have retry configured because the underlying method already has a retry mechanism. 

https://github.com/openwisp/openwisp-monitoring/blob/e563520c7765d4e638f8e65b8b8e4095fa2fc4cc/openwisp_monitoring/db/backends/influxdb/client.py#L194-L196